complete/_rg

```diff
@@ -87,6 +87,7 @@ _rg() {
     '(-w -x --line-regexp --word-regexp)'{-w,--word-regexp}'[only show matches surrounded by word boundaries]'
     '(-e -f --file --files --regexp --type-list)1: :_rg_pattern'
     '(--type-list)*:file:_files'
+    '*'{-z,--search-zip}'search through zip files'
```

Comment:
The exclusions and the syntax itself are wrong here. At a minimum it should be something like this:

```
'(-z --search-zip)'{-z,--search-zip}'[search through zip files]'
```

We might also want to add other exclusions, but i'd like to do another pass on this file within the next few weeks, so it can wait until then to get fancier.

I would also suggest changing the description wording to something like *search contents of compressed files* — 'through' is a bit ambiguous, and we don't actually search *zip* files at all.

Reply:
I'll fix the syntax as per your suggestion. This file is always throwing me off. Sorry about that.

I thought of using `-C/--search-compressed` instead of `-z/--search-zip`. The only reason I went with the latter is that it would be familiar to folks coming from using `ag`. Like you said, `search-zip` doesn't really make a lot of sense here. Should we just go with the former option?

Reply:
The completion-function syntax takes a while to get used to, yeah. Every few weeks i learn that something i used to do in these files is wrong, lol.

I like `-z` for the short option. It's what `ag` and `sift` both use, and the connection with 'Ziv' is immediately intuitive. As far as the long option, idk. `--search-zip` feels slightly inaccurate, but you're right about `ag` — it's one of the most common recursive file searchers, and definitely the most well known one that supports searching compressed files, so it's not unreasonable to borrow the terminology. `--search-compressed` makes sense too tho, w/e 🤷‍♀️